src/charm_state.py

```diff
@@ -0,0 +1,116 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""This module defines the CharmState class which represents the state of the Flask charm."""
+
+import datetime
+import pathlib
+
+from ops.charm import CharmBase
+
+from charm_types import WebserverConfig
+
+
+class CharmState:
+    """Represents the state of the Flask charm."""
+
+    def __init__(
+        self,
+        webserver_workers: int | None = None,
+        webserver_threads: int | None = None,
+        webserver_keepalive: int | None = None,
+        webserver_timeout: int | None = None,
+    ):
+        """Initialize a new instance of the CharmState class.
+
+        Args:
+            webserver_workers: The number of workers to use for the web server,
+                or None if not specified.
```

Comment:
Is it worth noting that it defaults to `None`? The function signature already shows this?

Reply:
Yes, I think the function signature already states that. And I think the second sentence implies that `None` is the default value.